File: resources/todo-js/get.js
// Run function for todo GET events using new context pattern
function Run(context) {
    context.log("Todo GET event executing with Run(context) pattern");
    
    // Add a computed field showing completion status
    context.data.status = context.data.completed ? "Done" : "Pending";
    
    // Format the creation date for display
    if (context.data.createdAt) {
        context.data.formattedDate = new Date(context.data.createdAt).toLocaleDateString();
    }
    
    // Add metadata about the processing
    context.data.processedBy = "JavaScript Run(context)";
    context.data.processedAt = new Date().toISOString();
}

File: resources/todo-go/get.go
import "time"

// Run processes todo data when retrieved
func Run(ctx *EventContext) error {
	// Add computed status field
	completed, _ := ctx.Data["completed"].(bool)
	if completed {
		ctx.Data["status"] = "Done"
	} else {
		ctx.Data["status"] = "Pending"
	}

	// Format dates for better display
	if createdAt, ok := ctx.Data["createdAt"].(time.Time); ok {
		ctx.Data["formattedDate"] = createdAt.Format("2006-01-02 15:04")
	}

	// Add priority label
	if priority, ok := ctx.Data["priority"].(float64); ok {
		switch int(priority) {
		case 1:
			ctx.Data["priorityLabel"] = "Low"
		case 2:
			ctx.Data["priorityLabel"] = "Normal"
		case 3:
			ctx.Data["priorityLabel"] = "High"
		case 4:
			ctx.Data["priorityLabel"] = "Urgent"
		case 5:
			ctx.Data["priorityLabel"] = "Critical"
		default:
			ctx.Data["priorityLabel"] = "Normal"
		}
	}

	return nil
}
